# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/waybar/hyprland.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Support hyprland workspaces.

  - NoTag <notag@maneto.fr>

--- Waybar-0.9.20/src/modules/wlr/workspace_manager.cpp.vanilla	2023-08-05 21:02:58.365131824 +0200
+++ Waybar-0.9.20/src/modules/wlr/workspace_manager.cpp	2023-08-05 21:03:02.025026792 +0200
@@ -523,7 +523,8 @@
   if (action.empty())
     return true;
   else if (action == "activate") {
-    zext_workspace_handle_v1_activate(workspace_handle_);
+    const std::string command = "hyprctl dispatch workspace " + name_;
+    system(command.c_str());
   } else if (action == "close") {
     zext_workspace_handle_v1_remove(workspace_handle_);
   } else {
